### ISSUE TITLE
8r1 - No Workie with WingX

### ISSUE THREAD

**Issue by ScraboTower** [cyoung/stratux#316]
1. Stratux version: 8R1
2. Stratux config:

    SDR
      [ ] single
      [X ] dual

    GPS
      [X ] yes
      [ ] no
      type:

    AHRS
      [X ] yes
      [ ] no

    power source: 

    usb cable:anker

3. EFB app and version: WingX 8.6.3.1

    EFB platform: iOS 9.2.1

    EFB hardware: iPhone 5s

4. Description of your issue:
Just updated using 8r1 image and there is no heartbeat to WingX but WingX shows it is connected to Stratux unit.  WebUi shows receivers etc are working

![stratux](https://cloud.githubusercontent.com/assets/15007252/13650600/ac229046-e600-11e5-9014-80a8a23c70f3.jpg)

If possible, enable "Replay Logs", reproduce the problem, and provide a copy of the logs in http://192.168.10.1/logs/stratux/ and http://192.168.10.1/logs/stratux.log.

**Comment by ScraboTower** [comment]
![stratux2](https://cloud.githubusercontent.com/assets/15007252/13650708/352e7e5e-e601-11e5-8879-d6a79b5db248.jpg)

**Comment by cyoung** [comment]
WingX version? I'll give it a try.


@HiltonG

**Comment by ScraboTower** [comment]
8.6.3.1   - there is nothing even on the Stratux tab under "External Hardware" either.

**Comment by AvSquirrel** [comment]
How did you upgrade to v0.8r1 -- .sh script, disk image, or pull/build?

I can't replicate on with 8.6.3.1 on iOS 9.2.1 and commit 796d0a263 made by `git checkout` and `make`.

**Comment by ScraboTower** [comment]
Image

**Comment by peepsnet** [comment]
Is this WingX on Apple?  I just bought WingX on android for $0.99 and activated the free trial. I cannot find any options for external hardware. So i returned the purchase.

**Comment by ScraboTower** [comment]
You mean this !!!

![Uploading WX1a.JPG…]()

**Comment by ScraboTower** [comment]
You mean you didn't see this  ??

![wx1a](https://cloud.githubusercontent.com/assets/15007252/13654451/a160f138-e615-11e5-94af-d5373e9eda61.JPG)

**Comment by emuyshondt** [comment]
I get similar results. Tried putting the 0.8r1 image on an SD card and didn't even get the stratux network to show up to connect my iPad Air 2, iOS 9.2.1. Then downloaded 0.7b1 img, and that worked. Updated to 0.8r1 using the update-stratux-v0.8r1-796d0a2632.sh file. 

I could then see the stratux network but even after an iPad reboot, I got an error on my iPad that says: Unable to join the network "stratux" 

My iPhone was able to connect, though. I see the stratux heartbeat in the latest WingX Pro, and everything seems to be working.

After playing with the phone for a couple of minutes, tried the iPad again and this time it connected and seems to work as expected.

Then I tried Foreflight. I saw some traffic. When I tapped it, FF gave a lot of information:
UPS2782 20,000' MSL 15.4nm S
Hdg: 210º Speed 316kts (978ADS-B)

I was surprised that what looks like a jet in the flight levels was using 978 ADS-B. I thought most of those would be using 1090ES transponders.

Henry

PS: My Pi works differently than under 0.6r1. Back then, there's a red LED that always stayed on on the Pi. My Pi had been left powered on for several weeks on my office desk, and was still working with the red LED on before this latest update. After installing 0.8 r1 and 0.7b1 img files, the red LED comes on at first, then a green LED starts flashing next to it, like a disk access light. The red LED goes out, but the green LED flashes for a bit. After a few seconds, all LEDs go out. It was kind of comforting before to see that something was going on. At first I thought my Pi had died. It took a little while for the Stratux network to show up after the light went out.

**Comment by skypuppy** [comment]
Isn't that a symptom of low power?

**Comment by peepsnet** [comment]
The red light not staying on is indeed a symptom of low power!

@ScraboTower  YUP!!!  Sure did. and the android version is behind. there seems to be no support for external devices yet

**Comment by peepsnet** [comment]
@emuyshondt Just redownloaded the img.zip
https://github.com/cyoung/stratux/releases/download/v0.8r1/stratux-v0.8r1-796d0a2632.img.zip

and it installed and booted with the stratux network working. all systems go for me

Try redownload and reimage

**Comment by AvSquirrel** [comment]
Interesting. The v0.8r1 image worked fine in WingX for about a minute, then quit. I rebooted and it came up and stayed up.

![img_0079](https://cloud.githubusercontent.com/assets/14127245/13655314/149a5aa0-e624-11e5-9d96-d50f6a0c6b30.PNG)

@cyoung -- are there any scripts or other configuration items in the image that get set on first boot? It's been a very long time since I've done anything other than git and build, and I've lost track.

@emuyshondt 
> I was surprised that what looks like a jet in the flight levels was using 978 ADS-B. I thought most of those would be using 1090ES transponders.

Known issue. Stratux uses the same byte format as the Garmin GDL90, which was a 978-only device. ForeFlight assumes any traffic sent with this format is 978/UAT, even if Stratux saw it on the 1090 receiver. 

There are two workarounds: Go to the Stratux web UI (http://192.168.10.1 and 1) look on the traffic page, or 2) turn on "Traffic Markings" on the settings UI page.

**Comment by Ergonomicmike** [comment]
Funny how the System Information in FF shows a second radio at 1090 but the app won't acknowledge that Traffic can come from 1090.

**Comment by ssokol** [comment]
I have a v0.8b1 system that was working fine all day with only my computer
as a client. Late in the day I connected Droid EFB and while I was able to
get GPS, I did not see any of the traffic targets that were listed in the
Traffic tab. I connected my iPhone and ForeFlight and it almost immediately
died - the connection to the access point remained, but no data was
flowing. I tried to ssh in and it hung. I tried ping and it gave no
response until, suddenly, it spit out a flood of responses all with nearly
30 seconds of lag. Then back to no response.

I rebooted it and as soon as it came back up the tablet and iPhone
reconnected. Within a few seconds it was in the same state. I finally shut
down the tablet and the iPhone and after 2 - 3 minutes the ping backlog
cleared. At that point the web interface began responding again and I was
able to ssh in. Everything seemed normal. It appears that the connection /
connections from EFBs triggers some kind of problem.

On Wed, Mar 9, 2016 at 7:52 PM, Ergonomicmike <notifications@github.com>
wrote:

> Funny how the System Information in FF shows a second radio at 1090 but
> the app won't acknowledge that Traffic can come from 1090.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyoung/stratux/issues/316#issuecomment-194615784>.
>



-- 
Steven Sokol
408 Camelot Drive
Liberty, MO 64068

mobile: +1 816-806-8844
fax: +1 816-817-0441

**Comment by ScraboTower** [comment]
Reimagined with 8r1 this morning.... Definately does not resume the heartbeat if WingX was suspended and resumed.

Restarted the Stratux but not WingX and WingX shows the wifi connection but no heartbeat.

**Comment by ScraboTower** [comment]
Went back to my previous installed version (v0.7b3-d714a5139e)  and its rock solid

**Comment by cyoung** [comment]
Thought at first it was the batched networking format - filling up the queue in sleep mode and bringing the app back doesn't cause it. Can't reproduce this.

@AvSquirrel - the first boot scripts are mostly things like setting an ssh key and such. Nothing stratux specific.

@ScraboTower - you didn't fill out the hardware profile in your original issue report. It looks like you're testing with an iPhone with at least iOS 7? I went back and edited your original issue report to include "WingX 8.6.3.1" so that this is easier to track.

@emuyshondt - are you using an iPhone as well?

**Comment by ScraboTower** [comment]
@cyoung   Yes I was using iPhone 5s with iOS 9.2.1.   Never used 8r1 against the iPad but can if that helps.

**Comment by cyoung** [comment]
I'm not sure if there are any relevant differences in iOS and/or WingX, it's worth a try if you're able to give v0.8r1 a shot on your iPad. I assume that everyone else is testing it with an iPad.

**Comment by jzeevi** [comment]
Do you mean between Android running Wingx? Or an iOS device running Wingx?

 

I would suspect that any differences seen between working and not working versions of Stratux (whatever flavor) eliminate the OS (Android or iOS).

**Comment by ScraboTower** [comment]
So updated to 8r1 using the .SH file this time, last time I used the image refresh (twice!!) 
90 mins later both iPad and iPhone are not showing the issue I reported. Even after switching wifi off and then back on again after a few mins, heartbeat resumed, same as resuming from having the app in the background.

**Comment by cyoung** [comment]
@ScraboTower - I think this bug may have not been WingX specific and https://github.com/cyoung/stratux/commit/b2eb382c1a27c7bae9774d712972701920a5ea64 may have fixed it.

Could you give v0.8r2 a try?

**Comment by emuyshondt** [comment]
I tried v0.8r2 today and I still saw an issue. I've updated my cellular iPar Air 2 to iOS 9.3, but I don't think the OS version has anything to do with this.

I still get a problem using the img file vs. updating 0.7b1 with the .sh file. 

My symptoms are this:

I loaded a new image 0.8r2 image onto my SD card. When I inserted it into my RPi, it came up, and I could see the Stratux network. However, when I tried to join with my iPad, I got an error that said: Unable to join the network "stratux". I tried rebooting the iPad, disabling WiFi on the iPad and then enabling it after a few minutes, but I could not connect. I also tried power cycling the RPi that is connected to a wall wart.

I then reimaged the SD card with version 0.7b1. With that image, I was able to connect to the stratux network. However, I wasn't able to update with the .sh file stored in my Box account. I went all the way through selecting a file in a folder, but immediately got an error that said: File not selected, and the UI froze. I couldn't tap on the OK button. I tried killing the browser session. When I restart it , I get the same problem. BTW, v0.8r2 now selects the file as expected. I didn't try actually installing it over itself, though.

I then went to my Mac and loaded the .sh update from there. Everything worked as expected. After reboot, I was able to join stratux with my iPad, and everything seems to be running well. I don't have enough technical skills to diagnose the connection issue.

One small nit: When I select the stratux network on my iPad, the WiFi connection symbol keeps cycling as when the iPad is trying to connect to a network. I know I'm connected because I get the UI, and I can see data from stratux, but the icon on the Mac (latest iOS) never stops moving. I guess it is a similar issue as the connected icon not showing on the iPad.

**Comment by Ergonomicmike** [comment]
In the middle of this thread https://www.reddit.com/r/stratux/comments/49josg/any_error_updating_to_v08r1/
someone wrote about the hoops he had to jump thru to get iOS to update via .sh and drop box. He says it's due to iOS paradigm.

**Comment by cyoung** [comment]
>One small nit: When I select the stratux network on my iPad, the WiFi connection symbol keeps cycling as when the iPad is trying to connect to a network. I know I'm connected because I get the UI, and I can see data from stratux, but the icon on the Mac (latest iOS) never stops moving. I guess it is a similar issue as the connected icon not showing on the iPad. 

Expected behavior.

**Comment by cyoung** [comment]
@emuyshondt @ScraboTower - have the issues persisted?

**Comment by ScraboTower** [comment]
@cyoung  - it has been fine with me since reinstall of 8R2

**Comment by emuyshondt** [comment]
I was able to get the file installed from Dropbox and Box after making a copy of the file available offline, per the Reddit thread. 



    _____________________________

**Comment by peepsnet** [comment]
If you did not make the file avaliable offline then it will try to pull it from the Internet but you are connected to stratux not the internet

**Comment by emuyshondt** [comment]
I was connected to the Internet via the cellular connection of my iPad but the file wouldn't download on its own. I had to make it available offline before it would work. I was able to navigate the directory structure to point to the file, so there were communications going on. It was not at the last step, where the actual file should transfer, that the process broke down.


    _____________________________
